### DISCUSSION TITLE
editmode in brick class

### DISCUSSION THREAD

**Discussion by BabSchu** [pimcore/pimcore#11308] [Q&A / Support] [answered]
Hello,

what is the best way to get access to editmode in brick classes in PimcoreX?
This is working, but seems like a poor man‘s solution:

`$info->getRequest()->attributes->get('_editmode');`

Using the EditmodeResolver on the other hand seems a bit long-winded to me. Thanks!

**Comment by leuchtdiode** [comment, marked as answer]
Would be interested too. Any suggestions @dvesh3?

BR,
Alex

**Reply by dvesh3** [reply]
why not simply inject editmodeResolver?
```
public function __construct(protected EditmodeResolver $editmodeResolver)
{
}

$this->editmodeResolver->isEditmode()
```

**Comment by leuchtdiode** [comment]
Alright, than we will use this class.

Thx & BR,
Alex
